contracts/two-party-pol-holder/src/contract.rs

```diff
@@ -194,32 +240,93 @@ fn try_claim(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Response, Con
+    pool: String,
+    mut covenant_config: TwoPartyPolCovenantConfig,
+) -> Result<Response, ContractError> {
+    // we figure out the amount of tokens to be expected
+    let entitled_assets: Vec<Asset> = deps.querier.query_wasm_smart(
+        pool.to_string(),
+        &astroport::pair::QueryMsg::Share {
+            amount: lp_token_bal,
+        },
+    )?;
+    // convert astro assets to coins
+    let mut withdraw_coins: Vec<Coin> = vec![];
+    for asset in entitled_assets {
+        withdraw_coins.push(asset.to_coin()?);
+    }
+
+    // generate the withdraw_liquidity hook for the claim party
+    let withdraw_liquidity_hook = &Cw20HookMsg::WithdrawLiquidity { assets: vec![] };
+    let withdraw_msg = &Cw20ExecuteMsg::Send {
+        contract: pool.to_string(),
+        amount: lp_token_bal,
+        msg: to_binary(withdraw_liquidity_hook)?,
+    };
+
+    let denom_splits = DENOM_SPLITS.load(deps.storage)?;
+    let mut distribution_messages: Vec<CosmosMsg> =
+        denom_splits.get_shared_distribution_messages(withdraw_coins);
+
+    // we submit the withdraw liquidity message followed by transfer of
+    // underlying assets to the corresponding router
+    let mut withdraw_and_forward_msgs = vec![CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: lp_token_addr.to_string(),
+        msg: to_binary(withdraw_msg)?,
+        funds: vec![],
+    })];
+
+    withdraw_and_forward_msgs.append(&mut distribution_messages);
+
+    claim_party.allocation = Decimal::zero();
+    counterparty.allocation = Decimal::zero();
+    covenant_config.update_parties(claim_party.clone(), counterparty);
+
+    // update the states
+    COVENANT_CONFIG.save(deps.storage, &covenant_config)?;
+    CONTRACT_STATE.save(deps.storage, &ContractState::Complete)?;
 
-    Ok(response.add_messages(withdraw_and_forward_msgs))
+    Ok(Response::default().add_messages(withdraw_and_forward_msgs))
 }
 
 fn try_tick(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Response, ContractError> {
     let state = CONTRACT_STATE.load(deps.storage)?;
     match state {
         ContractState::Instantiated => try_deposit(deps, env, info),
         ContractState::Active => check_expiration(deps, env),
-        ContractState::Expired => {
-            let config = COVENANT_CONFIG.load(deps.storage)?;
-            let state =
-                if config.party_a.allocation.is_zero() && config.party_b.allocation.is_zero() {
-                    CONTRACT_STATE.save(deps.storage, &ContractState::Complete)?;
-                    ContractState::Complete
-                } else {
-                    state
-                };
+        ContractState::Complete => Ok(Response::default()
+            .add_attribute("method", "tick")
+            .add_attribute("contract_state", state.to_string())),
+        // ragequit and expired
+        _ => {
+            let pool = POOL_ADDRESS.load(deps.storage)?;
+            let lp_token_bal = query_astro_pool_token(
```

Comment:
When is this called? seems like it will never be called

Reply:
Only in `Ragequit` state (in a share-based scenario), or when covenant is expired and claim is possible without ragequitting.
Do you think handling that case with `ContractState::Ragequit | ContractState::Expired => { ... }` would be better than `_ => { ... }`?

Reply:
100% yes, `_ =>` is handling "all the rest" while its working, you want to be explicit here because it should be handled only in those 2 cases right? 
If it really is meant to be handled in all other cases that are not specified then leave it as is, but if it is meant to be called ONLY for those 2 cases, then specify them.

The thing is, it gives you a type check when you add more states in the future, cause if you leave it as is, and add a new state, you will not see any error given, and you might forget to handle it here.
where if you do specify those 2 cases here, it would give you an error when adding a new state, which tells you to handle it.

Reply:
Great point, added that!